implementations/micrometer-registry-dynatrace/src/test/java/io/micrometer/dynatrace/v2/DynatraceExporterV2Test.java

```diff
@@ -0,0 +1,404 @@
+        assertThat(actual.get(0)).startsWith("my.function.timer,dt.metrics.source=micrometer gauge,min=10.0,max=10.0,sum=5005.0,count=500 ");
+        assertThat(actual.get(0)).hasSize("my.function.timer,dt.metrics.source=micrometer gauge,min=10.0,max=10.0,sum=5005.0,count=500 1234567890123".length());
+    }
+
+    @Test
+    void toGaugeInvalidName() {
+        // invalid name.
+        meterRegistry.gauge("~~~", 1.23);
+        Gauge gauge = meterRegistry.find("~~~").gauge();
+        assertNotNull(gauge);
+
+        List<String> actual = exporter.toGaugeLine(gauge).collect(Collectors.toList());
+        assertThat(actual).isEmpty();
+    }
+
+    @Test
+    void toGaugeInvalidCases() {
+        meterRegistry.gauge("my.gauge1", Double.NaN);
+        Gauge gauge1 = meterRegistry.find("my.gauge1").gauge();
+        assertNotNull(gauge1);
+        List<String> actual1 = exporter.toGaugeLine(gauge1).collect(Collectors.toList());
+        assertThat(actual1).isEmpty();
+
+        meterRegistry.gauge("my.gauge2", Double.NEGATIVE_INFINITY);
+        Gauge gauge2 = meterRegistry.find("my.gauge2").gauge();
+        assertNotNull(gauge2);
+        List<String> actual2 = exporter.toGaugeLine(gauge2).collect(Collectors.toList());
+        assertThat(actual2).isEmpty();
+
+        meterRegistry.gauge("my.gauge3", Double.POSITIVE_INFINITY);
+        Gauge gauge3 = meterRegistry.find("my.gauge3").gauge();
+        assertNotNull(gauge3);
+        List<String> actual3 = exporter.toGaugeLine(gauge3).collect(Collectors.toList());
+        assertThat(actual3).isEmpty();
+    }
+
+    @Test
+    void toGaugeTags() {
+        // invalid name.
+        Gauge.builder("my.gauge", () -> 1.23).tags(Tags.of("tag1", "value1", "tag2", "value2")).register(meterRegistry);
+        Gauge gauge = meterRegistry.find("my.gauge").gauge();
+        assertNotNull(gauge);
+
+        List<String> actual = exporter.toGaugeLine(gauge).collect(Collectors.toList());
+        assertThat(actual).hasSize(1);
+        assertThat(actual.get(0)).contains("tag1=value1").contains("tag2=value2").contains("dt.metrics.source=micrometer");
+        assertThat(actual.get(0)).startsWith("my.gauge,");
+        assertThat(actual.get(0)).hasSize("my.gauge,dt.metrics.source=micrometer,tag1=value1,tag2=value2 gauge,1.23 1617776498381".length());
+    }
+
+    @Test
+    void toCounterInvalidName() {
+        // invalid name.
+        meterRegistry.counter("~~~");
+        Counter counter = meterRegistry.find("~~~").counter();
+        assertNotNull(counter);
+
+        List<String> actual = exporter.toCounterLine(counter).collect(Collectors.toList());
+        assertThat(actual).isEmpty();
+    }
+
+    @Test
+    void toCounterInvalidCases() {
+        meterRegistry.counter("my.counter1");
+        Counter counter = meterRegistry.find("my.counter1").counter();
+        assertNotNull(counter);
+
+        // NaN and infinity are ignored. The counter value stays at 0.0 when adding one of NaN,
```

Comment:
@pirgeo You see this behavior because there is a step counter under the hood that returns you the previous value (`0.0`) until the next step.